File: code_samples/scala/networking_tcp_udp.scala
/**
 * Networking: Tcp Udp
 * AI/ML Training Sample
 */

class TcpUdp {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object TcpUdpExample {
  def main(args: Array[String]): Unit = {
    val instance = new TcpUdp()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/networking_tcp_udp.hs
-- Networking: Tcp Udp
-- AI/ML Training Sample

module TcpUdp where

data TcpUdp = TcpUdp {
    getData :: String
} deriving (Show, Eq)

process :: TcpUdp -> String -> TcpUdp
process obj input = obj { getData = input }

validate :: TcpUdp -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = TcpUdp { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
